include/pybind11/iostream.h

```diff
@@ -43,11 +43,15 @@ class pythonbuf : public std::streambuf {
     // simplified to a fully qualified call.
     int _sync() {
         if (pbase() != pptr()) {
-            // This subtraction cannot be negative, so dropping the sign
-            str line(pbase(), static_cast<size_t>(pptr() - pbase()));
 
             {
                 gil_scoped_acquire tmp;
+
+                // This subtraction cannot be negative, so dropping the sign
+                // this invokes python and therefore should be included in the
+                // GIL
```

Comment:
Punctuation. Just try reading this comment as a single long sentence.

Reply:
Good point.

Reply:
It should be better now